plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -519,7 +519,12 @@ export class PersonState {
 
                 let personOverrideMessages: ProducerRecord[] = []
                 if (this.poEEmbraceJoin) {
-                    personOverrideMessages = [await this.addPersonOverride(otherPerson, mergeInto, tx)]
+                    personOverrideMessages = await new PersonOverrideWriter(this.db.postgres).addPersonOverride(
```

Comment:
I take it the coming change to have multiple different `PersonOverrideWriter`s (even one that writes to both?) will mean this `new Foo` gets pulled up elsewhere? Only asking because `await new Foo()` looks funny compared to just calling a plain fn.

Reply:
Yeah, the specific `PersonOverrideWriter` implementation will get switched between this one or the new deferred implementation (which shares the same interface) based on some runtime setting.

What I'm planning:

- `PersonOverrideWriter`: this one, does what it already does here
- `DeferredPersonOverrideWriter`: new one
  - `addPersonOverride` writes to the pending person override table instead
  - exposes some method to allow for running the queue consumer/worker that reads from the pending person override table; this will use the `PersonOverrideWriter` to maintain the (non-pending) person overrides table (and mappings) until we're ready to do the table migration

This could all be done with plain functions too, I just find the relationships between objects a bit easier to keep in my head/more descriptive than functions wrapping other functions — either approach works though, I'm not strongly wedded to/against either one.

Reply:
Yep, makes sense. Honestly I have to untrain my brain from C++ PTSD about allocating an object just to throw it away.